common-ktx/src/main/java/co/nimblehq/common/extensions/TypedArrayExt.kt

```diff
@@ -0,0 +1,63 @@
+package co.nimblehq.common.extensions
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.content.res.TypedArray
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import android.os.Build
+import androidx.annotation.FontRes
+import androidx.annotation.StyleableRes
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.content.res.ResourcesCompat
+
+/**
+ * Unfortunatelly we need to pass context, as there is no proper way to parse font from fonts resource.
+ * There is variant to parse raw resource and get it as InputStream, then convert it into file and then pass to Typeface.
+ * But it is much easier to pass context.
+ */
+fun TypedArray.getTypeface(context: Context, @StyleableRes index: Int) =
+    runCatching { getFontCompat(context, index) }.getOrNull() ?: getString(index)?.let {
+        Typeface.create(
+            it,
+            Typeface.NORMAL
+        )
+    }
+
+fun TypedArray.getFontCompat(context: Context, @StyleableRes index: Int): Typeface? {
```

Comment:
The `index` is really vague IMO, I think you could add the sample of `index` for usage.